src/main/scala/chisel3/tester/ChiselScalatestTester.scala

```diff
@@ -2,24 +2,28 @@
 
 package chisel3.tester
 
-import firrtl.ExecutionOptionsManager
 import chisel3.experimental.MultiIOModule
-import chisel3.tester.internal.{BackendInstance, Context, FailedExpectException, TestEnvInterface, TesterOptions}
+import chisel3.tester.internal._
+import firrtl.AnnotationSeq
 import org.scalatest._
 import org.scalatest.exceptions.TestFailedException
 
 import scala.util.DynamicVariable
 
 trait ChiselScalatestTester extends Assertions with TestSuiteMixin with TestEnvInterface { this: TestSuite =>
-  class TestBuilder[T <: MultiIOModule](val dutGen: () => T,
-      val execOptions: Option[ExecutionOptionsManager], val testOptions: Option[TesterOptions]) {
-    protected def getTestOptions: TesterOptions = {
-      val test = scalaTestContext.value.get
-      TesterOptions(test.name, test.configMap.contains("writeVcd"))
+  class TestBuilder[T <: MultiIOModule](val dutGen: () => T, annotationSeq: AnnotationSeq) {
+    def getTestName: String = {
+      scalaTestContext.value match {
+        case Some(x) =>
+          sanitizeFileName(x.name)
+        case _ => "a"
```

Comment:
will use `.get`. I tend to think of a naked `.get` as a bit sketchy, the exception thrown by that is unlikely to be of any use to an end user, but it is probably not that likely to happen here.

Reply:
This use of `"a"` may have come from me originally (see: `AddImplicitOutputFile` phase that adds an output file annotation derived from a circuit, falling back to `"a"` if no such circuit exists). This was a legacy carry-over of the idea of "viewing" an arbitrary, possibly garbage `AnnotationSeq` as some type where you had to provide a default for a non-optional value. But, the "view" idea predates Stage/Phase. Once you move to Stage/Phase, then you have an explicit ordering of Phases (enforced/canonicalized by the dependency API) and can rely on guarantees on what is in the annotation sequence.

So, if you can guarantee that this will always succeed, a naked `.get` should be fine. If you can't guarantee that, then it might hint that there's a phase-based organization that could untangle things.

Reply:
Unless there's a known or expected failure mode, I'd prefer things fail noisily so it won't result in unexpected (and therefore difficult-to-debug) behavior.

We might also want to have a better discussion of robust error handling, but my current philosophy is that the stack trace of a .get() is going to provide enough information for a developer to debug it. And users that encounter a random stack trace probably are just going to open an issue and post the stack trace, I don't expect them to be able to do anything with a random failure in the middle of the compilation process.

Reply:
Yeah, I think a hard error here is fine. This error is entirely unexpected and a stack trace is better than the alternative.

Towards error handling (not a solution to accumulate errors), we could just use what @jackkoenig has done with [FIRRTL error handling](https://github.com/freechipsproject/firrtl/blob/master/src/main/scala/firrtl/stage/FirrtlStage.scala#L27) everywhere to push users to file issues on unexpected exceptions.

Reply:
Made it into a naked `.get`